src/components/common/ColonyMembers/MembersFilter/MembersFilter.tsx

```diff
@@ -0,0 +1,192 @@
+    defaultMessage: 'Banned',
+  },
+  notBanned: {
+    id: `${displayName}.notBanned`,
+    defaultMessage: 'Not banned',
+  },
+  memberType: {
+    id: `${displayName}.memberType`,
+    defaultMessage: 'Member type',
+  },
+  bannedStatus: {
+    id: `${displayName}.bannedStatus`,
+    defaultMessage: 'Banned status',
+  },
+  verificationType: {
+    id: `${displayName}.verificationType`,
+    defaultMessage: 'Verification type',
+  },
+});
+
+export enum MemberType {
+  ALL = 'all',
+  CONTRIBUTORS = 'contributors',
+  WATCHERS = 'watchers',
+}
+
+export enum VerificationType {
+  ALL = 'all',
+  VERIFIED = 'verified',
+  UNVERIFIED = 'unverified',
+}
+
+export enum BannedStatus {
+  ALL = 'all',
+  BANNED = 'banned',
+  NOT_BANNED = 'not_banned',
+}
+
+export interface FormValues {
+  memberType: MemberType;
+  verificationType: VerificationType;
+  bannedStatus: BannedStatus;
+}
+
+const memberTypes = [
+  { label: MSG.allMembers, value: MemberType.ALL },
+  { label: MSG.contributors, value: MemberType.CONTRIBUTORS },
+  { label: MSG.watchers, value: MemberType.WATCHERS },
+];
+
+const verificationTypes = [
+  { label: MSG.any, value: VerificationType.ALL },
+  { label: MSG.verified, value: VerificationType.VERIFIED },
+  { label: MSG.unverified, value: VerificationType.UNVERIFIED },
+];
+
+const bannedStatuses = [
+  { label: MSG.any, value: BannedStatus.ALL },
+  { label: MSG.banned, value: BannedStatus.BANNED },
+  { label: MSG.notBanned, value: BannedStatus.NOT_BANNED },
+];
+
+interface Props {
+  handleFiltersCallback: (filters: FormValues) => void;
+  isRoot: boolean;
+}
+
+const MembersFilter = ({ handleFiltersCallback, isRoot }: Props) => {
+  const selectRef = useRef<HTMLDivElement>(null);
+
+  const scrollIntoView = () => {
+    selectRef.current?.scrollIntoView({ behavior: 'smooth' });
+  };
+
+  return (
+    <>
+      <hr className={styles.divider} />
+      <Form
+        initialValues={{
+          memberType: MemberType.ALL,
+          verificationType: VerificationType.ALL,
+          bannedStatus: BannedStatus.ALL,
+        }}
+        onSubmit={() => {}}
+        enableReinitialize
+      >
+        {({ resetForm, values }: FormikProps<FormValues>) => {
+          handleFiltersCallback(values);
+          return (
+            <div className={styles.filters}>
+              <div className={styles.titleContainer}>
+                <span className={styles.title}>
+                  <FormattedMessage {...MSG.filter} />
+                </span>
+                {(values.bannedStatus !== BannedStatus.ALL ||
+                  values.verificationType !== VerificationType.ALL ||
+                  values.memberType !== MemberType.ALL) && (
+                  <Button
+                    text={MSG.reset}
+                    appearance={{ theme: 'blue' }}
+                    onClick={() => resetForm()}
+                  />
+                )}
+              </div>
+              {isRoot && (
+                <Select
```

Comment:
Given that these `Selects` are all the same view, we could represent each `Select` as a data object, and then map through an array of these to produce the filters list. 

E.g. 

``` js

filters.map(({name, options, label}) => 
  <Filter name={name} options={options} label={label}/>
)

```

I think this array should live in the file with the arrays/enums above.

Reply:
This is the pattern I'm proposing: 

Config : https://github.com/safe-global/web-core/blob/dev/src/components/sidebar/SidebarNavigation/config.tsx

View: https://github.com/safe-global/web-core/blob/dev/src/components/sidebar/SidebarNavigation/index.tsx